Failed: SessionPoolStubbedSpec

```
info] - may reject non-serializable events (40 milliseconds)
[info] + CapabilityFlag `supportsSerialization` was turned `on`.  
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-12] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://JournalSpec/temp/$0b#0] in-use Map() availableQueue(AvailableSession(Session(projects/akka/instances/akka/databases/SpannerJournalSpec/sessions/3,Map(),Some(Timestamp(1587473704,489545000,UnknownFieldSet(Map()))),Some(Timestamp(1587473704,489546000,UnknownFieldSet(Map()))),UnknownFieldSet(Map())),1587473706384))
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://JournalSpec/temp/$1b#0] in-use Map(58bcc2d0-8c5a-4a5d-b9ae-5030d78d98a7 -> Session(projects/akka/instances/akka/databases/SpannerJournalSpec/sessions/3,Map(),Some(Timestamp(1587473704,489545000,UnknownFieldSet(Map()))),Some(Timestamp(1587473704,489546000,UnknownFieldSet(Map()))),UnknownFieldSet(Map()))) availableQueue()
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://JournalSpec/temp/$2b#0] in-use Map(58bcc2d0-8c5a-4a5d-b9ae-5030d78d98a7 -> Session(projects/akka/instances/akka/databases/SpannerJournalSpec/sessions/3,Map(),Some(Timestamp(1587473704,489545000,UnknownFieldSet(Map()))),Some(Timestamp(1587473704,489546000,UnknownFieldSet(Map()))),UnknownFieldSet(Map()))) availableQueue()
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://JournalSpec/temp/$3b#0] in-use Map(58bcc2d0-8c5a-4a5d-b9ae-5030d78d98a7 -> Session(projects/akka/instances/akka/databases/SpannerJournalSpec/sessions/3,Map(),Some(Timestamp(1587473704,489545000,UnknownFieldSet(Map()))),Some(Timestamp(1587473704,489546000,UnknownFieldSet(Map()))),UnknownFieldSet(Map()))) availableQueue()
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - ReleaseSession 58bcc2d0-8c5a-4a5d-b9ae-5030d78d98a7
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://JournalSpec/temp/$1b#0] in-use Map() availableQueue(AvailableSession(Session(projects/akka/instances/akka/databases/SpannerJournalSpec/sessions/3,Map(),Some(Timestamp(1587473704,489545000,UnknownFieldSet(Map()))),Some(Timestamp(1587473704,489546000,UnknownFieldSet(Map()))),UnknownFieldSet(Map())),1587473706404))
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-12] a.p.s.i.SessionPool$ [] - ReleaseSession 4912a3fc-a424-41d3-ba32-d6a53984ba18
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-12] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://JournalSpec/temp/$2b#0] in-use Map() availableQueue(AvailableSession(Session(projects/akka/instances/akka/databases/SpannerJournalSpec/sessions/3,Map(),Some(Timestamp(1587473704,489545000,UnknownFieldSet(Map()))),Some(Timestamp(1587473704,489546000,UnknownFieldSet(Map()))),UnknownFieldSet(Map())),1587473706407))
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - ReleaseSession ded90695-4d9c-43f0-a370-08961fbcbc14
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://JournalSpec/temp/$3b#0] in-use Map() availableQueue(AvailableSession(Session(projects/akka/instances/akka/databases/SpannerJournalSpec/sessions/3,Map(),Some(Timestamp(1587473704,489545000,UnknownFieldSet(Map()))),Some(Timestamp(1587473704,489546000,UnknownFieldSet(Map()))),UnknownFieldSet(Map())),1587473706410))
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-12] a.p.s.i.SessionPool$ [] - ReleaseSession a0288588-778e-498a-9b48-2a3b5e14a0b3
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://JournalSpec/temp/$4b#0] in-use Map() availableQueue(AvailableSession(Session(projects/akka/instances/akka/databases/SpannerJournalSpec/sessions/3,Map(),Some(Timestamp(1587473704,489545000,UnknownFieldSet(Map()))),Some(Timestamp(1587473704,489546000,UnknownFieldSet(Map()))),UnknownFieldSet(Map())),1587473706413))
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-12] a.p.s.i.SessionPool$ [] - ReleaseSession 8c8eaea6-3a65-4aff-b0be-8d9542415626
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://JournalSpec/temp/$5b#0] in-use Map() availableQueue(AvailableSession(Session(projects/akka/instances/akka/databases/SpannerJournalSpec/sessions/3,Map(),Some(Timestamp(1587473704,489545000,UnknownFieldSet(Map()))),Some(Timestamp(1587473704,489546000,UnknownFieldSet(Map()))),UnknownFieldSet(Map())),1587473706421))
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://JournalSpec/temp/$6b#0] in-use Map(9ad22ba4-2e6b-48fc-852e-0ea9fdc21bda -> Session(projects/akka/instances/akka/databases/SpannerJournalSpec/sessions/3,Map(),Some(Timestamp(1587473704,489545000,UnknownFieldSet(Map()))),Some(Timestamp(1587473704,489546000,UnknownFieldSet(Map()))),UnknownFieldSet(Map()))) availableQueue()
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - ReleaseSession 9ad22ba4-2e6b-48fc-852e-0ea9fdc21bda
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://JournalSpec/temp/$6b#0] in-use Map() availableQueue(AvailableSession(Session(projects/akka/instances/akka/databases/SpannerJournalSpec/sessions/3,Map(),Some(Timestamp(1587473704,489545000,UnknownFieldSet(Map()))),Some(Timestamp(1587473704,489546000,UnknownFieldSet(Map()))),UnknownFieldSet(Map())),1587473706429))
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-12] a.p.s.SpannerInteractions [SpannerInteractions(akka://JournalSpec)] - Single row. 6
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-12] a.p.s.i.SessionPool$ [] - ReleaseSession a7a30023-f7b5-48b0-a504-dfe36bfe1f72
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-12] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://JournalSpec/temp/$7b#0] in-use Map() availableQueue(AvailableSession(Session(projects/akka/instances/akka/databases/SpannerJournalSpec/sessions/3,Map(),Some(Timestamp(1587473704,489545000,UnknownFieldSet(Map()))),Some(Timestamp(1587473704,489546000,UnknownFieldSet(Map()))),UnknownFieldSet(Map())),1587473706434))
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-12] a.p.s.i.SessionPool$ [] - ReleaseSession c78b9ca3-61b0-450a-9dcb-2aa447be2053
[info] - may serialize events (42 milliseconds)
04/21 12:55:06 INFO [JournalSpec-akka.actor.default-dispatcher-9] a.p.s.SpannerSpec [SpannerSpec(akka://JournalSpec)] - Database dropped SpannerJournalSpec
[info] - should deletes sessions on shutdown *** FAILED *** (3 seconds, 54 milliseconds)
[info]   java.lang.AssertionError: Timeout (3 seconds) during expectMessageClass waiting for class akka.persistence.spanner.internal.SessionPoolStubbedSpec$DeleteSessionInvocation
[info]   at akka.actor.testkit.typed.internal.TestProbeImpl.assertFail(TestProbeImpl.scala:386)
[info]   at akka.actor.testkit.typed.internal.TestProbeImpl.expectMessageClass_internal(TestProbeImpl.scala:233)
[info]   at akka.actor.testkit.typed.internal.TestProbeImpl.expectMessageType(TestProbeImpl.scala:212)
[info]   at akka.persistence.spanner.internal.SessionPoolStubbedSpec$$anon$7.$anonfun$new$9(SessionPoolStubbedSpec.scala:157)
[info]   at scala.collection.immutable.List.foreach(List.scala:392)
[info]   at akka.persistence.spanner.internal.SessionPoolStubbedSpec$$anon$7.<init>(SessionPoolStubbedSpec.scala:156)
[info]   at akka.persistence.spanner.internal.SessionPoolStubbedSpec.$anonfun$new$8(SessionPoolStubbedSpec.scala:152)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.wordspec.AnyWordSpecLike$$anon$3.apply(AnyWordSpecLike.scala:1076)
[info]   at org.scalatest.TestSuite.withFixture(TestSuite.scala:196)
[info]   at org.scalatest.TestSuite.withFixture$(TestSuite.scala:195)
[info]   at akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit.withFixture(ScalaTestWithActorTestKit.scala:29)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.invokeWithFixture$1(AnyWordSpecLike.scala:1074)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTest$1(AnyWordSpecLike.scala:1086)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTest(AnyWordSpecLike.scala:1086)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTest$(AnyWordSpecLike.scala:1068)
[info]   at akka.persistence.spanner.internal.SessionPoolStubbedSpec.runTest(SessionPoolStubbedSpec.scala:65)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$runTests$1(AnyWordSpecLike.scala:1145)
[info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:413)
[info]   at scala.collection.immutable.List.foreach(List.scala:392)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:390)
[info]   at org.scalatest.SuperEngine.$anonfun$runTestsInBranch$1(Engine.scala:427)
[info]   at scala.collection.immutable.List.foreach(List.scala:392)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.runTestsInBranch(Engine.scala:396)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:475)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTests(AnyWordSpecLike.scala:1145)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.runTests$(AnyWordSpecLike.scala:1144)
[info]   at akka.persistence.spanner.internal.SessionPoolStubbedSpec.runTests(SessionPoolStubbedSpec.scala:65)
[info]   at org.scalatest.Suite.run(Suite.scala:1112)
[info]   at org.scalatest.Suite.run$(Suite.scala:1094)
[info]   at akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit.org$scalatest$BeforeAndAfterAll$$super$run(ScalaTestWithActorTestKit.scala:29)
[info]   at org.scalatest.BeforeAndAfterAll.liftedTree1$1(BeforeAndAfterAll.scala:213)
[info]   at org.scalatest.BeforeAndAfterAll.run(BeforeAndAfterAll.scala:210)
[info]   at org.scalatest.BeforeAndAfterAll.run$(BeforeAndAfterAll.scala:208)
[info]   at akka.persistence.spanner.internal.SessionPoolStubbedSpec.org$scalatest$wordspec$AnyWordSpecLike$$super$run(SessionPoolStubbedSpec.scala:65)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.$anonfun$run$1(AnyWordSpecLike.scala:1190)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:535)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.run(AnyWordSpecLike.scala:1190)
[info]   at org.scalatest.wordspec.AnyWordSpecLike.run$(AnyWordSpecLike.scala:1188)
[info]   at akka.persistence.spanner.internal.SessionPoolStubbedSpec.run(SessionPoolStubbedSpec.scala:65)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:318)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:513)
[info]   at sbt.TestRunner.runTest$1(TestFramework.scala:139)
[info]   at sbt.TestRunner.run(TestFramework.scala:154)
[info]   at sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.$anonfun$apply$1(TestFramework.scala:317)
[info]   at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:277)
[info]   at sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:317)
[info]   at sbt.TestFramework$$anon$3$$anonfun$$lessinit$greater$1.apply(TestFramework.scala:317)
[info]   at sbt.TestFunction.apply(TestFramework.scala:329)
[info]   at sbt.Tests$.$anonfun$toTask$1(Tests.scala:311)
[info]   at sbt.std.Transform$$anon$3.$anonfun$apply$2(Transform.scala:46)
[info]   at sbt.std.Transform$$anon$4.work(Transform.scala:67)
[info]   at sbt.Execute.$anonfun$submit$2(Execute.scala:281)
[info]   at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:19)
[info]   at sbt.Execute.work(Execute.scala:290)
[info]   at sbt.Execute.$anonfun$submit$1(Execute.scala:281)
[info]   at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:178)
[info]   at sbt.CompletionService$$anon$2.call(CompletionService.scala:37)
[info]   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[info]   at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[info]   at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[info]   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[info]   at java.base/java.lang.Thread.run(Thread.java:834)
04/21 12:55:07 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-3] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$h] - Creating pool. Max size [2]. Stash [1].
04/21 12:55:07 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-3] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$h] - Creating sessions projects/akka/instances/akka/databases/akka 2
04/21 12:55:07 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$h] - Sessions created BatchCreateSessionsResponse(List(Session(s1,Map(),None,None,UnknownFieldSet(Map())), Session(s2,Map(),None,None,UnknownFieldSet(Map()))),UnknownFieldSet(Map()))
04/21 12:55:07 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://SessionPoolStubbedSpec/system/testProbe-74#-1278477261] in-use Map() availableQueue(AvailableSession(Session(s1,Map(),None,None,UnknownFieldSet(Map())),1587473707234), AvailableSession(Session(s2,Map(),None,None,UnknownFieldSet(Map())),1587473707234))
04/21 12:55:07 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-8] a.p.s.i.SessionPool$ [] - ReleaseSession 125e3fa2-08b1-403f-9863-e4b0f29beff9
04/21 12:55:07 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-8] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://SessionPoolStubbedSpec/system/testProbe-74#-1278477261] in-use Map() availableQueue(AvailableSession(Session(s2,Map(),None,None,UnknownFieldSet(Map())),1587473707234), AvailableSession(Session(s1,Map(),None,None,UnknownFieldSet(Map())),1587473707239))
04/21 12:55:07 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - ReleaseSession 05832b3a-c0ac-4520-a5e8-7ec5a8a089a3
04/21 12:55:07 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://SessionPoolStubbedSpec/system/testProbe-74#-1278477261] in-use Map() availableQueue(AvailableSession(Session(s1,Map(),None,None,UnknownFieldSet(Map())),1587473707239), AvailableSession(Session(s2,Map(),None,None,UnknownFieldSet(Map())),1587473707241))
04/21 12:55:07 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-8] a.p.s.i.SessionPool$ [] - ReleaseSession 3f6d4017-866e-4ef9-82c1-ab7e49c0ff93
[info] - should returns sessions in round robin order (16 milliseconds)
04/21 12:55:07 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$i] - Creating pool. Max size [2]. Stash [1].
04/21 12:55:07 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$i] - Creating sessions projects/akka/instances/akka/databases/akka 2
04/21 12:55:07 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-8] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$i] - Sessions created BatchCreateSessionsResponse(List(Session(s1,Map(),None,None,UnknownFieldSet(Map())), Session(s2,Map(),None,None,UnknownFieldSet(Map()))),UnknownFieldSet(Map()))
04/21 12:55:08 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-8] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$h] - The following sessions haven't been used in the last 1.000 s. Sending keep alive. Queue(Session(s2,Map(),None,None,UnknownFieldSet(Map())), Session(s1,Map(),None,None,UnknownFieldSet(Map())))
04/21 12:55:08 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-8] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$i] - The following sessions haven't been used in the last 1.000 s. Sending keep alive. Queue(Session(s1,Map(),None,None,UnknownFieldSet(Map())), Session(s2,Map(),None,None,UnknownFieldSet(Map())))
[info] - should keeps alive un-used sessions (1 second, 25 milliseconds)
04/21 12:55:08 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-3] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$j] - Creating pool. Max size [2]. Stash [1].
04/21 12:55:08 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-3] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$j] - Creating sessions projects/akka/instances/akka/databases/akka 2
04/21 12:55:08 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-3] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$j] - Sessions created BatchCreateSessionsResponse(List(Session(s1,Map(),None,None,UnknownFieldSet(Map())), Session(s2,Map(),None,None,UnknownFieldSet(Map()))),UnknownFieldSet(Map()))
04/21 12:55:09 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$j] - The following sessions haven't been used in the last 1.000 s. Sending keep alive. Queue(Session(s1,Map(),None,None,UnknownFieldSet(Map())), Session(s2,Map(),None,None,UnknownFieldSet(Map())))
04/21 12:55:09 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-3] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://SessionPoolStubbedSpec/system/testProbe-78#-25303515] in-use Map(ff14680b-70af-4369-85fa-c525a44b7bf1 -> Session(s1,Map(),None,None,UnknownFieldSet(Map())), a1264d4a-6faa-43a7-afa0-b6bbd1bde0b4 -> Session(s2,Map(),None,None,UnknownFieldSet(Map()))) availableQueue()
04/21 12:55:09 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-3] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://SessionPoolStubbedSpec/system/testProbe-78#-25303515] in-use Map(ff14680b-70af-4369-85fa-c525a44b7bf1 -> Session(s1,Map(),None,None,UnknownFieldSet(Map())), a1264d4a-6faa-43a7-afa0-b6bbd1bde0b4 -> Session(s2,Map(),None,None,UnknownFieldSet(Map()))) availableQueue()
04/21 12:55:09 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - ReleaseSession ff14680b-70af-4369-85fa-c525a44b7bf1
04/21 12:55:09 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://SessionPoolStubbedSpec/system/testProbe-78#-25303515] in-use Map(a1264d4a-6faa-43a7-afa0-b6bbd1bde0b4 -> Session(s2,Map(),None,None,UnknownFieldSet(Map()))) availableQueue(AvailableSession(Session(s1,Map(),None,None,UnknownFieldSet(Map())),1587473709404))
04/21 12:55:09 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - ReleaseSession a1264d4a-6faa-43a7-afa0-b6bbd1bde0b4
04/21 12:55:09 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-9] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://SessionPoolStubbedSpec/system/testProbe-78#-25303515] in-use Map(2393b645-f1d8-4be4-befc-9125b47df4cf -> Session(s1,Map(),None,None,UnknownFieldSet(Map()))) availableQueue(AvailableSession(Session(s2,Map(),None,None,UnknownFieldSet(Map())),1587473709405))
[info] - should sessions should not be available during keep alive (1 second, 126 milliseconds)
04/21 12:55:09 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-3] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$k] - Creating pool. Max size [2]. Stash [1].
04/21 12:55:09 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-3] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$k] - Creating sessions projects/akka/instances/akka/databases/akka 2
04/21 12:55:09 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-8] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$k] - Sessions created BatchCreateSessionsResponse(List(Session(s1,Map(),None,None,UnknownFieldSet(Map())), Session(s2,Map(),None,None,UnknownFieldSet(Map()))),UnknownFieldSet(Map()))
04/21 12:55:10 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-8] a.p.s.i.SessionPool$ [akka://SessionPoolStubbedSpec/user/$k] - The following sessions haven't been used in the last 1.000 s. Sending keep alive. Queue(Session(s1,Map(),None,None,UnknownFieldSet(Map())), Session(s2,Map(),None,None,UnknownFieldSet(Map())))
04/21 12:55:10 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-8] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://SessionPoolStubbedSpec/system/testProbe-80#851090695] in-use Map(be3f0faa-fb3d-4949-b46a-f248786c699d -> Session(s1,Map(),None,None,UnknownFieldSet(Map())), 27e83bba-f786-4c12-b69d-b92cbc15de1d -> Session(s2,Map(),None,None,UnknownFieldSet(Map()))) availableQueue()
04/21 12:55:10 WARN [SessionPoolStubbedSpec-akka.actor.default-dispatcher-8] a.p.s.i.SessionPool$ [] - failed to keep session alive, may be re-tried again before expires server side
akka.persistence.spanner.internal.SessionPoolStubbedSpec$$anon$11$$anon$12: oh dear
04/21 12:55:10 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-8] a.p.s.i.SessionPool$ [] - ReleaseSession be3f0faa-fb3d-4949-b46a-f248786c699d
04/21 12:55:10 INFO [SessionPoolStubbedSpec-akka.actor.default-dispatcher-8] a.p.s.i.SessionPool$ [] - GetSession from Actor[akka://SessionPoolStubbedSpec/system/testProbe-80#851090695] in-use Map(27e83bba-f786-4c12-b69d-b92cbc15de1d -> Session(s2,Map(),None,None,UnknownFieldSet(Map()))) availableQueue(AvailableSession(Session(s1,Map(),None,None,UnknownFieldSet(Map())),1587473710532))
```

https://travis-ci.com/github/akka/akka-persistence-spanner/jobs/321623974

--- a/journal/src/test/scala/akka/persistence/spanner/internal/SessionPoolStubbedSpec.scala
+++ b/journal/src/test/scala/akka/persistence/spanner/internal/SessionPoolStubbedSpec.scala
@@ -152,11 +152,12 @@ class SessionPoolStubbedSpec extends ScalaTestWithActorTestKit with Matchers wit
     "deletes sessions on shutdown" in new Setup {
       val response = probe.expectMessageType[BatchSessionCreateInvocation].response
       response.complete(Success(BatchCreateSessionsResponse(sessions)))
+      pool ! GetSession(sessionProbe.ref, id1)
+      sessionProbe.expectMessageType[PooledSession]
       testKit.stop(pool)
-      sessions.foreach { session =>
-        val request = probe.expectMessageType[DeleteSessionInvocation]
-        request.input shouldEqual DeleteSessionRequest(session.name)
-      }
+      sessions.map { _ =>
+        probe.expectMessageType[DeleteSessionInvocation].input.name
+      }.toSet shouldEqual sessions.map(_.name).toSet
     }
 
     "returns sessions in round robin order" in new Setup {

